### PR TITLE
Clean up move panel code for air transport dependent units.

### PR DESCRIPTION
**Pull request by asvitkine**
## Change Summary & Additional Notes

Clean up move panel code for air transport dependent units.

The previous code was unnecessarily complex given dependentUnits was tracking only the loaded units for the current move, so they only need to be cleared when the move is issued or canceled.

Also cleans up a few things in related code, such as unnecessary conditionals that are always true.

No functional changes.

Tested by loading air transports and landing them, undoing moves and canceling a move in progress.

### CODE REVIEW

**Comment by DanVanAtta** [comment]
@asvitkine , wanted to give a quick update - I'm pretty slammed. I hope to have some time for TripleA in the next coming two to four days.

**Comment by asvitkine** [comment]
> @asvitkine , wanted to give a quick update - I'm pretty slammed. I hope to have some time for TripleA in the next coming two to four days.

No worries, no rush. But if you think someone else would be a good reviewer, you can also suggest. But no problem to wait a bit.

**Comment by DanVanAtta** [comment]
@asvitkine - I believe code reviews should be more 'peer reviews'. Basically a reviewer should mostly simulate a future maintainer and check for clarity and reasonableness. Essentially, does someone other than the author, or the author many weeks/months down the road have a good chance to be able to easily enough understand and modify the code? For this, anyone can really do the review.

There are places where checking in that the design fits with overall strategy, is consistent, etc... those are places where it is good to have specific people to give feedback. The other area is for UI and major features, there it is just good to have a reasonable second or third opinion.

With the above said, I think you have demonstrated that your concerns are in the right place, eg: correctness and save game compatibility.

So, I invite you to have write permissions @asvitkine . For anything that is more minor, pretty short, and where you are sure about correctness, you can mostly just merge those changes. I try to leave changes open for a good 12-24 hours at least if I can and often a person can avoid the rush, come back, do a self review and merge. Otherwise, if it is something that is less trivial, it's good to try and leave the PR open for 2~5 days  if possible, then do a self review and merge if nobody is able to review. UI and functionality changes probably should generally be reviewed, but there are judgment calls to be made with that.

> But if you think someone else would be a good reviewer, you can also suggest. But no problem to wait a bit.

@RoiEXLab generally watches out for mentions and certainly can jump in. It is desired, and a goal, for us to have an increased pool of reviewers, right now that is very scarce.

**Comment by RoiEXLab** [comment]
> @RoiEXLab generally watches out for mentions and certainly can jump in. It is desired, and a goal, for us to have an increased pool of reviewers, right now that is very scarce.

Unfortunately my freetime is very inconsistent. There are weeks like this one, where I can afford to spend quite some time here, but the majority of weeks my time is rather limited if I want to have some IRL social life :D 
So if there's a pressing issue, as @DanVanAtta already said, I'd be happy to jump in but for common changes it depends and might take a day or two before I can have a look.

**Comment by asvitkine** [comment]
> @asvitkine - I believe code reviews should be more 'peer reviews'. Basically a reviewer should mostly simulate a future maintainer and check for clarity and reasonableness. Essentially, does someone other than the author, or the author many weeks/months down the road have a good chance to be able to easily enough understand and modify the code? For this, anyone can really do the review.
> 
> There are places where checking in that the design fits with overall strategy, is consistent, etc... those are places where it is good to have specific people to give feedback. The other area is for UI and major features, there it is just good to have a reasonable second or third opinion.

I think there's one other aspect - when changing something tricky, it can be worthwhile for a reviewer familiar with that part of the code to double check that the assumptions are correct. On the other hand, I suspect a lot of the codebase is effectively legacy code such that there may not be a reviewer that knows it deeply. I agree with everything else, though.

> 
> With the above said, I think you have demonstrated that your concerns are in the right place, eg: correctness and save game compatibility.
> 
> So, I invite you to have write permissions @asvitkine . For anything that is more minor, pretty short, and where you are sure about correctness, you can mostly just merge those changes. I try to leave changes open for a good 12-24 hours at least if I can and often a person can avoid the rush, come back, do a self review and merge. Otherwise, if it is something that is less trivial, it's good to try and leave the PR open for 2~5 days if possible, then do a self review and merge if nobody is able to review. UI and functionality changes probably should generally be reviewed, but there are judgment calls to be made with that.

Sounds good, will do. I think in practice, I do value code reviews so I'll still wait for a review for most changes, but agree that trivial ones or ones where I got an approval with some minor comments, I could just commit directly.

**Review by DanVanAtta — APPROVED** [review]

**Pull request merged**
